2022/incoming/time-to-open-presents.pod

```diff
@@ -0,0 +1,191 @@
+Author: oodler@cpan.org
+Title: Wake up! Time to open presents! 
+Topic: $SIG::ALRM, signal handlers, alarm , Try::ALRM
+
+=encoding utf8
+
+=head1 Wake up! Time to open presents!
+
+Clark W. Grisman faced a dilemma, it was the night before
+Christmas; and he found himself up against a programming
+deadline. Why a company at the cutting edge of milk proof
+cereal coatings required him to write a Perl program,
+Clark can't remember. But he does know that he's trying
+to read from a socket that keeps timing out! 
+
+=head1 Traditional C<alarm> Handling
+
+In the past whenever Clark wanted to call a procedure inside
+his Perl code that might take too long to return, he always
+simply copied the example in C<perldoc -f alarm>, which at 
+the time of this Advent season looks like the following:
+
+=begin perl
+
+# ...
+
+eval {
+    local $SIG{ALRM} = sub { die "alarm\n" }; # NB: \n required
+    alarm $timeout;
+    my $nread = sysread $socket, $buffer, $size;
+    alarm 0;
+};
+if ($@) {
+    die unless $@ eq "alarm\n";   # propagate unexpected errors
+    # timed out
+}
+else {
+    # didn't
+}
+
+=end perl
+
+Or if he didn't need it quite so fancy, he'd just nip it down to:
+
+=begin perl
+
+$SIG{ALRM} = sub { die "alarm\n" }; # NB: \n required
+alarm $timeout;
+my $nread = sysread $socket, $buffer, $size;
+alarm 0;
+
+=end perl
+
+Clark never liked using C<alarm> in this latter way because it didn't feel
+very I<perlish>. But he did what he had to, then would just bottle
+up the guilt. That's what the eggnog is for, he'd tell himself.
+
+=head1 C<alarm> Handling Comes to Christmastown
+
+Clark is not only a food scientist by day, but he considers
+himself a Perl artist by night. Realizing this approach seemed
+a little dated, he recalled seeing a module fly by the CPAN radar
+that treats C<alarm>'s C<ALRM> signal as a catchable exception.
+
+After some searching, he found the module. L<Try::ALRM>. Turns
+out there was no need to fiddle with the  C<eval> stuff directly,
+using this handy module it can be done like the following using
```

Comment:
```suggestion
out there was no need to fiddle with the  C<eval> stuff directly.
Using this handy module, it can be done in the following way using
```